src/OSPSuite.Presentation/Presenters/SimulationOutputMappingPresenter.cs

```diff
@@ -127,24 +140,64 @@ private IEnumerable<DataRepository> allAvailableObservedDataUsedBySimulation()
       public void UpdateSimulationOutputMappings(SimulationOutputMappingDTO simulationOutputMappingDTO)
       {
          MarkSimulationAsChanged();
+
+         if (simulationOutputMappingDTO.Output.DisplayString.Equals(Captions.SimulationUI.NoneEditorNullText))
+         {
+            removeOutputMapping(simulationOutputMappingDTO);
```

Comment:
I absolutely don't get it. Why would we have to remove a NULL entry from the simulation? This does not make sense to me

Reply:
when the output is set to null, the outputMapping should not be in the simulation (if you remember we have already discussed about this), so we have to explicitely delete it in order not to have OutputMappings with null outputs in the simulation.

Reply:
true, setting the DTO output to null is not anymore necessary (come sfrom the way the code was before), so I am removing. But about removing the outputMapping from the Simulation, why not? The Presenter was doing it till now anyway (just in a different place, when hitting the "x" button). I mean, it is the SimulationOutputMappingPresenter, and removing it from the simulation is an one-liner, I definately do not see a need for a service or something like that here.

Reply:
moving the OutputMapping removal code to the simulation itself.